fdbclient/GlobalConfig.actor.cpp

```diff
@@ -237,56 +240,65 @@ ACTOR Future<Void> GlobalConfig::refresh(GlobalConfig* self) {
 // Applies updates to the local copy of the global configuration when this
 // process receives an updated history.
 ACTOR Future<Void> GlobalConfig::updater(GlobalConfig* self, const ClientDBInfo* dbInfo) {
-	wait(self->cx->onConnected());
-	wait(self->migrate(self));
-
-	wait(self->refresh(self));
-	self->initialized.send(Void());
-
 	loop {
 		try {
-			wait(self->dbInfoChanged.onTrigger());
+			wait(self->cx->onConnected());
+			wait(self->migrate(self));
 
-			auto& history = dbInfo->history;
-			if (history.size() == 0) {
-				continue;
-			}
+			wait(self->refresh(self));
+			self->initialized.send(Void());
```

Comment:
I think this will fail if you've already executed this line once before. You could probably put all 4 of these lines above in an `if (self->initialized.canBeSet())` block.